projects/step-frontend/src/lib/modules/timeseries/dashboard/grouping/ts-grouping.component.html

```diff
@@ -0,0 +1,29 @@
+<button mat-button [matMenuTriggerFor]="groupingMenu" #matTrigger="matMenuTrigger">
+  Grouping: {{ selectedGrouping.label }}
+</button>
+<mat-menu #groupingMenu="matMenu">
+  <button
+    mat-menu-item
+    *ngFor="let grouping of groupingOptions"
+    (click)="selectedGrouping = grouping; emitGroupingChange()"
+  >
+    {{ grouping.label }}
+  </button>
+  <div mat-menu-item [matMenuTriggerFor]="customGroupingMenu">Other</div>
+  <mat-menu #customGroupingMenu="matMenu">
+    <div (click)="$event.stopPropagation()" style="padding: 12px">
```

Comment:
please avoid inline styles

Reply:
Personally, I prefer to have inline style for situations like this, where there is a single element which needs a small padding or margin. Several people online sustain this approach, so I don't think it is definitely something bad. I see this thing easier to follow, less code, less classes, faster to render. I think it is more of a preference thing though, but if you are all against it, I will definitely align. Just le me know! @neogucky @dvladir  @bojidaryovchev